src/components/FilterInput.svelte

```diff
@@ -16,14 +18,17 @@
   }
   input {
     width: 100%;
+    cursor: pointer;
+    box-shadow: $box-shadow-sm;
   }
 </style>
 
 <div>
   <input
-    type="text"
+    placeHolder={text}
+    style="border-radius: 15px; border: 1px solid #000;"
```

Comment:
This should be in the css class above, no?

Reply:
The reason for inline styling was because I wanted to override Protocol rules in class selectors: https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity, but I realized just now that I can also achieve this by giving them an id

Reply:
I think in this case we should use `!important` to override the default behaviour. I know this is normally discouraged but I think we're following the rules of thumb outlined here:

https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_!important_exception

Reply:
I'm having reservations about this, mostly because we need to override Protocol quite often (due to it being compiled last in the process hence always has higher priority over Svelte component css), and using `!important` across the the stylesheet like that in the long term would make it harder to debug.

> I know this is normally discouraged but I think we're following the rules of thumb outlined here:
> 
> [https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_!important_exception](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_!important_exception)

Rule of thumb number 1 is "Always look for a way to use specificity before even considering !important", and since id selector is included in the docs as one of selector types recommended when we need to use specificity: https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#selector_types and https://developer.mozilla.org/en-US/docs/Web/CSS/ID_selectors, I don't think we're straying from convention by using ids. I don't have a strong attachment to this though, let me know what you think.

Reply:
Here's a resource which suggests that id's should be unique (which more or less corresponds to what I've seen in the wild): https://css-tricks.com/the-difference-between-id-and-class/#ids-are-unique

*However*, I agree that using ids is probably preferably to `!important` for the reasons you mention. I assume that option (2) under the [important exception](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_!important_exception) ("use more specific rules") wouldn't work for some reason?

Reply:
> However, I agree that using ids is probably preferably to !important for the reasons you mention. I assume that option (2) under the important exception ("use more specific rules") wouldn't work for some reason?

Yes I tried it, but even with more specific targeting, it would still end up being overridden by Protocol (if limited to type and class selectors) because Protocol rules are compiled last so they show up last on the style sheet. The only way to counter this is either using ids or `!important`. What I'm worried about `!important` is that since the need to customize styling is quite frequent, and if we decide to use `!important` pattern whenever we need to customize, it can get messy really fast. If everything is important, what's actually important?

> Here's a resource which suggests that id's should be unique (which more or less corresponds to what I've seen in the wild): https://css-tricks.com/the-difference-between-id-and-class/#ids-are-unique

I will make sure that the ids are a) unique (which is usually the case because when I want to override Protocol I aim at a very specific element) and b) used sparingly, only when absolutely needed.